### PR TITLE
Button to test a workflow action

### LINKED ISSUE #3745: Test Workflow Button
### Is your proposal related to a problem?

A user does not see what the reminder template or their custom template actually looks like in an email or SMS message. 

### Describe the solution you'd like

Add  a 'Test workflow' button, that sends a test Email/SMS to the user

### LINKED ISSUE #3963: mobile workflow is not responsive with sms reminder
see twist

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Adds a button to test a workflow action by sending out an email or SMS. Additionally, some design improvements have been made.

Also fixes: #3963

Old design: 
Screenshot 2022-08-16 at 15 11 11

New design: 
Screenshot 2022-08-16 at 15 10 41

Fixes #3745

 Loom Video: https://www.loom.com/share/f3db501b545a41cc8807e267f886d0f5

**Environment**: Staging(main branch) / Production

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

- Setup Sendgrid and Twilio to test locally: https://github.com/calcom/cal.com#workflows
- Create a workflow
- Choose any action (except send SMS to attendee)
- Click ‘Test Action’
- Choose reminder template or create your own custom template
- Check if you received the correct SMS or Email

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/features/ee/workflows/components/WorkflowStepContainer.tsx [hunk 1]
```diff
@@ -515,6 +549,29 @@ export default function WorkflowStepContainer(props: WorkflowStepProps) {
                   )}
                 </>
               )}
+              {form.getValues(`steps.${step.stepNumber - 1}.action`) !== WorkflowActions.SMS_ATTENDEE && (
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/components/WorkflowStepContainer.tsx:552]: We don't allow testing the action ‘send SMS to attendee’ as we do not want to send SMS to random attendees from a booking. If a user wants to see how the SMS template looks like, they need to switch to ‘Send SMS to specific number`

File: packages/features/ee/workflows/components/WorkflowStepContainer.tsx [hunk 2]
```diff
@@ -515,6 +549,29 @@ export default function WorkflowStepContainer(props: WorkflowStepProps) {
...
+                      form.getValues(`steps.${step.stepNumber - 1}.action`) !== WorkflowActions.SMS_NUMBER
+                    ) {
+                      testActionMutation.mutate({
+                        action: step.action,
+                        emailSubject: step.emailSubject || "",
+                        reminderBody: step.reminderBody || "",
+                        template: step.template,
+                      });
+                    } else {
+                      setConfirmationDialogOpen(true);
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/components/WorkflowStepContainer.tsx:568]: Asking for additional confirmation if user really wants to send SMS to the given number

File: packages/features/ee/workflows/pages/workflow.tsx [hunk 3]
```diff
@@ -88,7 +88,7 @@ function WorkflowPage() {
   });
 
   useEffect(() => {
-    if (workflow) {
+    if (workflow && !form.getValues("name")) {
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/pages/workflow.tsx:91]: This caused some wrong updates of the form values

**Review by zomars — APPROVED** [review]
Nice work @CarinaWolli Thank you 🙏🏽 

Computer Reaction GIF by GitHub

**Pull request merged**